File: montecarlo.py
import random

def estimate_pi(num_samples):
    inside_circle = 0

    for _ in range(num_samples):
        x = random.uniform(0, 1)
        y = random.uniform(0, 1)

        if x**2 + y**2 <= 1:
            inside_circle += 1

    return 4 * inside_circle / num_samples

# Example run
samples = 1000000
result = estimate_pi(samples)
print(f"Estimated value of π using {samples} samples: {result}")

File: monte carlo/montecarlo.js
function estimatePi(numPoints) {
  let insideCircle = 0;

  for (let i = 0; i < numPoints; i++) {
    const x = Math.random();
    const y = Math.random();

    if (x * x + y * y <= 1) {
      insideCircle++;
    }
  }

  const piEstimate = (insideCircle / numPoints) * 4;
  return piEstimate;
}

// Example: Run with 1 billion points
const numPoints = 1_000_000_000;
console.time("Monte Carlo Pi");
const pi = estimatePi(numPoints);
console.timeEnd("Monte Carlo Pi");

console.log(`Estimated Pi with ${numPoints.toLocaleString()} points:`, pi);
